### PR TITLE
Fix test_math refcounting error

### PR DESCRIPTION
**Pull request by Daetalus**

### CODE REVIEW

**Comment by undingen** [comment]
:+1:

File: src/runtime/float.cpp [hunk 1]
```diff
@@ -163,7 +163,10 @@ extern "C" double PyFloat_AsDouble(PyObject* o) noexcept {
         return -1;
     }
 
-    return static_cast<BoxedFloat*>(fo)->d;
+    val = static_cast<BoxedFloat*>(fo)->d;
+    Py_DECREF(fo);
```
Comment (kmod) [inline src/runtime/float.cpp]: Hi, shouldn't this get covered by the AUTO_DECREF on line 160?

Comment (undingen) [inline src/runtime/float.cpp]: oh yeah you are right :blush:

Comment (Daetalus) [inline src/runtime/float.cpp]: Sorry for the mistake... :worried:

Comment (kmod) [inline src/runtime/float.cpp]: It's not really your fault if we have the tests turned off :P (they are on now)

**Pull request merged**